### ISSUE TITLE
Add option to embed chat as well as video

### ISSUE THREAD

**Issue by psteger** [gfscott/eleventy-plugin-embed-twitch#13]
Ideally, I was thinking of adding chat alongside the video stream. My thinking here is to add a `chatWindow: false` to the pluginDefaults file and use that to show the chat only if it's true to preserve backward compatibility for sites currently using the plugin.  There are 2 ways to accomplish this that I can see if we have `chatWindow: true`:

1. Change buildEmbed to use the Twitch.Embed object passing `video-with-chat` in the layout parameter. This looks **much** cleaner and I personally prefer this route but there's a slight chance of breaking backward compatibility going this route. See here: [https://dev.twitch.tv/docs/embed/everything](https://dev.twitch.tv/docs/embed/everything)

2. Add a second iframe that contains the chat window in buildEmbed. This gives the ability to place it in different locations than the default right-hand side that Twitch streams usually have it and it's guaranteed to preserve backward compatibility. This feels almost like it should be a separate plugin if it goes this route to allow for the most flexibility in placement though. See here:[https://dev.twitch.tv/docs/embed/chat](https://dev.twitch.tv/docs/embed/chat)

Thoughts/suggestions?

**Comment by gfscott** [comment]
(We talked about this briefly by email but let's do it again for the record! 😜)

I'm into this! My priority is to keep the default behavior of the plugin as straightforward as possible — paste URL, get embed — and offer more advanced functionality by configuration. So this totally feels in-scope.

I agree that the first option seems more appealing. I'd probably tackle it something like this:

- Refactor [lib/buildEmbed.js](https://github.com/gfscott/eleventy-plugin-embed-twitch/blob/main/lib/buildEmbed.js) to accommodate multiple embedding strategies. That's how I handled it [in the YouTube plugin](https://github.com/gfscott/eleventy-plugin-youtube-embed/blob/main/lib/buildEmbed.js) and a couple months on I'm feeling like that approach works, and is reasonably future-friendly.
  - **Strategy 1**: Keep the existing iframe embed, which remains the default behavior
  - **Strategy 2**: Add a _new_ behavior that uses `Twitch.Embed`. It can be a script-based option for users who prefer that to iframes. They have the option of implementing it with or without `video-with-chat`.
- Chat is _only_ available if you're using Strategy 2. (Since this is advanced functionality, that feels like a fair tradeoff to me, but curious if you agree!) it also avoids breaking backward compatibility.
- You could add a simple convenience option like `chat: true` which would automatically force a switch to Strategy 2 with some sensible defaults.

Anyway, great idea! Very happy to review a PR or collaborate on a branch — whatever works for you 😀

**Comment by psteger** [comment]
I get the `chat: true` convenience option, but the `lite` option you used in the YouTube plugin doesn't feel right here. What do you think of `version` or `embedVersion`? In terms of implementation, everything you've specified looks solid and easy enough to hash out.

**Comment by gfscott** [comment]
Yeah, good point, `chat: true` is getting a little too cute, let's drop that idea.

Could we go with `embedMethod` for the option key? Accepting a value of either `iframe` (default) or `script`? (I don't have as strong an opinion on `script`, perhaps you have an alternative.)

**Comment by psteger** [comment]
It's been a heck of a week since I last pinged you. I'm cool with `embedMethod`. I kind of like `chat` but maybe `video-with-chat` is better since it keeps the language the same as Twitch? I'm wide open on that.

I like `iframe` for the default option key but I'd prefer `twitchEmbed` or `twitch-embed` for the new style. `twitch-embed` again keeps things 1 to 1 with the language Twitch is using. 

I'm going to work on this tonight and any changes we make to the language should just be a simple find and replace before I do a commit/PR for you to review.
